tesseract/tesseract_planning/tesseract_process_managers/src/taskflow_generators/graph_taskflow.cpp

```diff
@@ -192,16 +192,25 @@ tf::Taskflow& GraphTaskflow::generateTaskflow(ProcessInput input,
 
 void GraphTaskflow::abort()
 {
+  abort_ = true;
   for (auto& node : nodes_)
     node.process->setAbort(true);
 }
 
 void GraphTaskflow::reset()
 {
+  abort_ = false;
```

Comment:
This can segfault. 
```
 for (auto& node : nodes_)
    if (node.process)
       node.process->setAbort(false);

```

Reply:
Perhaps this check should go in the addNode. It came about because I was doing this.
```
  auto contact_check_generator = std::make_unique<DiscreteContactCheckProcessGenerator>();
  int contact_check_idx_1 = graph->addNode(std::move(contact_check_generator), GraphTaskflow::NodeType::CONDITIONAL);
  int contact_check_idx_2 = graph->addNode(std::move(contact_check_generator), GraphTaskflow::NodeType::CONDITIONAL);
```

It seems odd conceptually that the same generator can't be passed into multiple nodes.

Reply:
Now that I think about it, is it ever possible that it would be a null pointer. It is a unique pointer which this class owns so if it is in the vector of nodes it should have a valid pointer. If it does not, then this class deleted it which should not happen.

Reply:
Is there some reason addNode has to take a unique pointer? Or could it take a reference? The Node stores a ProcessGenerator::UPtr, but there shouldn't be any reason multiple nodes couldn't use the same ProcessGenerator. That is sort of the purpose of the generators.

Reply:
> Perhaps this check should go in the addNode. It came about because I was doing this.
> 
> ```
>   auto contact_check_generator = std::make_unique<DiscreteContactCheckProcessGenerator>();
>   int contact_check_idx_1 = graph->addNode(std::move(contact_check_generator), GraphTaskflow::NodeType::CONDITIONAL);
>   int contact_check_idx_2 = graph->addNode(std::move(contact_check_generator), GraphTaskflow::NodeType::CONDITIONAL);
> ```
> 
> It seems odd conceptually that the same generator can't be passed into multiple nodes.

The main reason was the ability to cancel things. If they were added to two different taskflows and you called cancel in one it would kill the processes in the other taskflow. We could switch it back to Unique Pointers and leave it up to the developer to make sure they do not do this.

Reply:
Now that things are working we can look at restructuring to see what should be unique. It was easier to do this to start to insure that I did not run into race conditions while trouble shooting the how the data flows through everything.

Reply:
@mpowelson If we could change the ProcessGenerator to where it returns an object that gets stored at the higher level this would allows for the same process generator to be used.

Reply:
Maybe what should happen is that the process generators should clone and store all of the information they need when generate is called. e.g. clone the planner. As for the abort, I think it is reasonable that any process created from the same generator shares an abort flag, but that's just my opinion.